Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py

```diff
@@ -2416,11 +2416,49 @@ def run_snippet_code_script_command(client: CoreClient, args: Dict) -> CommandRe
     )
 
 
+def form_powershell_command(unescaped_string: str):
+    """
+    Builds a Powershell command using prefix and a shell-escaped string.
+
+    Args:
+        unescaped_string (str): An unescaped command string.
+
+    Returns:
+        str: Prefixed and escaped command.
+    """
+    escaped_string = ''
+    for i in range(len(unescaped_string)):
+        char = unescaped_string[i]
+
+        if char == "'":
+            escaped_string += "''"
+
+        elif char == '"':
+            backslash_count = 0
+            for j in range(i - 1, -1, -1):
+                if unescaped_string[j] != '\\':
+                    break
+                backslash_count += 1
+
+            escaped_string += ('\\' * backslash_count) + '\\"'
+
+        else:
+            escaped_string += char
+
+    return f"powershell -Command '{escaped_string}'"
+
+
 def run_script_execute_commands_command(client: CoreClient, args: Dict) -> CommandResults:
     endpoint_ids = argToList(args.get('endpoint_ids'))
     incident_id = arg_to_number(args.get('incident_id'))
     timeout = arg_to_number(args.get('timeout', 600)) or 600
-    parameters = {'commands_list': argToList(args.get('commands'))}
+
+    commands_string = args.get('commands', '')
+    commands: list = [commands_string] if args.get('is_raw_command') else argToList(commands_string)
```

Comment:
```suggestion
    commands_string = args.get('commands')
    commands: list = [commands_string] if argToBoolean(args.get('is_raw_command')) else argToList(commands_string)
```
Notice that in case the `commands_string` is `None` or `''`, the value `argToList` returns is an empty list, but in case `is_raw_command` is True the result will be `[None]` or `['']`.
![image](https://github.com/user-attachments/assets/4ccb7284-e2fa-4003-af65-0032bd9a4f2d)
I think it is better not to add a default value here because the `commands` argument is required and the behavior in case the `commands` list has the value of `['']` is not defined well, but it is up to you.

Reply:
@BEAdi Thank you for the helpful feedback :)

Fixed in commit fc8aa2dc9fcaade623cfa4823b48e39f64f501aa:
* Wrapped the `is_raw_command` arg by `argToBoolean` to avoid cases like `"false"` (string) being detected as `True`.
* Used `remove_empty_elements` for the `commands` arg to avoid the `[None]` case you described.